deployment/terraform/variables.tf

```diff
@@ -189,3 +189,54 @@ variable "rds_free_memory_threshold_bytes" {
 variable "rds_cpu_credit_balance_threshold" {
   default = "30"
 }
+
+variable "ec2_service_role_policy_arn" {
+  default = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"
+}
+
+variable "batch_service_role_policy_arn" {
+  default = "arn:aws:iam::aws:policy/service-role/AWSBatchServiceRole"
+}
+
+variable "spot_fleet_service_role_policy_arn" {
+  default = "arn:aws:iam::aws:policy/service-role/AmazonEC2SpotFleetTaggingRole"
+}
+
+variable "batch_default_ce_spot_fleet_bid_percentage" {
+  default = "40"
+}
+
+variable "batch_ami_id" {
+  # Latest ECS-optimized Amazon Linux AMI in us-east-1
+  # See: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html
+  default = "ami-0bf2fb355727b7faf"
+}
+
+variable "batch_default_ce_min_vcpus" {
+  default = "0"
+}
+
+variable "batch_default_ce_desired_vcpus" {
+  default = "0"
+}
+
+variable "batch_default_ce_max_vcpus" {
+  default = "16"
+}
+
+variable "batch_default_ce_instance_types" {
+  type = "list"
+
+  default = [
+    "c5",
```

Comment:
Do you think the address matching algorithms are going to need instances this large, at least from the start? What do you think about starting with `t3`?

Also, I'm curious why we're only specifying the instance family. How does Batch determine which specific instance type to use?

Reply:
> What do you think about starting with `t3`?

I think `t3` is reasonable to start with, especially on staging. Later iterations of the address matching algorithm may need more compute depending on how complicated the approach becomes. (I've leaned towards `m5` instances for big dedupe jobs in the past.) I didn't actually know this Batch setup was for the deduplication process, though -- these values were just copied over from the Raster Foundry Batch setup.

> How does Batch determine which specific instance type to use?

The [docs](https://docs.aws.amazon.com/batch/latest/userguide/compute_environment_parameters.html) don't explain in detail, but based on anectodal experience it seems like Batch selects the instance type within the family that meets your desired vCPU count. With a desired count of 2 I've gotten exclusively `m5.large` instances, but the one time I accidentally set the desired count to 512 (wrong units!) I got a bunch of `m5.24xlarge` instances that each spun up for a few seconds. I'm not quite sure what the specific algorithm is for fulfilling the desired count, though (e.g. whether it will prioritize larger instances, or cheaper purchases).

Reply:
Thank you for your detailed answers to my questions.

I think that using `m3` instances as we discussed out-of-band on Slack makes sense.

I also think your anecdotal experience about how Batch selects the instance type makes sense too.

🙆‍♂️